NumLib/Assembler/VectorMatrixAssembler.h

```diff
@@ -95,17 +95,14 @@ class VectorMatrixAssembler<GlobalMatrix, GlobalVector, LocalAssembler,
         const double t, GlobalVector const& x,
         GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b) const
     {
-        auto cb = [&local_assembler](
-                std::vector<double> const& local_x,
-                LocalToGlobalIndexMap::RowColumnIndices const& r_c_indices,
-                const double t,
-                GlobalMatrix& M, GlobalMatrix& K, GlobalVector& b)
-        {
-            local_assembler.assemble(t, local_x);
-            local_assembler.addToGlobal(r_c_indices, M, K, b);
-        };
-
-        passLocalVector_(cb, id, _data_pos, x, t, M, K, b);
+        auto const x_and_indices = prepareLocalVector(id, _data_pos, x);
```

Comment:
I don't need an explicit indices vector in prepareLocalVector(), but I need to know about the element's indices there. So the indices vector could be omitted in prepareLocalVector(), but then I'd need to query the dof table twice for the element's indices.
Furthermore the local DOFs and the corresponding indices are linked. Computing them at the same spot emphasizes that, and makes sure they always match (I admit, the chance of a mismatch is not very high).
I'd like to keep the return value of prepareLocalVector() as it is.